test/support/pulp/task_support.rb

```diff
@@ -26,6 +26,7 @@ def self.wait_on_tasks(task_list, options = {})
 
     def self.any_task_running(async_tasks)
       async_tasks.each do |t|
+        next if t.finished?
```

Comment:
this avoids re-polling tasks that are already finished, such as if 1 out of 3 are finished, just poll the two that aren't.